gravitee-am-test/specs/gateway/forgot-password.jest.spec.ts

```diff
@@ -0,0 +1,464 @@
+                "minLength": 5,
+                "maxLength": 24,
+                "includeNumbers": true,
+                "includeSpecialCharacters": true,
+                "lettersInMixedCase": true,
+                "maxConsecutiveLetters": 5,
+                "expiryDuration": 9,
+                "passwordHistoryEnabled": true,
+                "oldPasswords": 3
+            }
+        }
+    },
+    {
+        inherited: true,
+        domain: {
+            accountSettings: {
+                "resetPasswordCustomForm": true,
+                "resetPasswordCustomFormFields": [
+                    {
+                        "key": "email",
+                        "label": "Email",
+                        "type": "email"
+                    },
+                    {
+                        "key": "username",
+                        "label": "Username",
+                        "type": "text"
+                    }
+                ],
+                "resetPasswordConfirmIdentity": true,
+                "resetPasswordInvalidateTokens": true
+            },
+            passwordSettings: {
+                "minLength": 5,
+                "maxLength": 64,
+                "includeNumbers": true,
+                "includeSpecialCharacters": true,
+                "lettersInMixedCase": true,
+                "maxConsecutiveLetters": 5,
+                "expiryDuration": 9,
+                "passwordHistoryEnabled": true,
+                "oldPasswords": 3
+            }
+        }
+    }
+];
+
+const selectSetting = (setting) => {
+    return setting.inherited ? setting.domain : setting.app;
+};
+
+const expectedMsg = (setting) => {
+    return selectSetting(setting).accountSettings.autoLoginAfterResetPassword ? selectSetting(setting).accountSettings.redirectUriAfterResetPassword : "success=reset_password_completed";
+}
+
+settings.forEach(setting => {
+    const selectedSetting = selectSetting(setting);
+    describe("Gateway reset password", () => {
+        beforeAll(async () => {
+            const adminTokenResponse = await requestAdminAccessToken();
+            accessToken = adminTokenResponse.body.access_token;
+            domain = await createDomain(accessToken, faker.company.companyName(0), "test Gateway reset password with password history");
+            domainIds.add(domain.id);
+            domain = await patchDomain(domain.id, accessToken, {
+                "master": true,
+                "oidc": {
+                    "clientRegistrationSettings": {
+                        "allowLocalhostRedirectUri": true,
+                        "allowHttpSchemeRedirectUri": true,
+                        "allowWildCardRedirectUri": true
+                    }
+                },
+                "loginSettings": {
+                    "forgotPasswordEnabled": true,
+                },
+                "accountSettings": selectedSetting.accountSettings,
+                "passwordSettings": selectedSetting.passwordSettings
```

Comment:
account & password Settings are initialized using the `selectedSettings` object but this object contains either the settings for the domain or the app. Here you assign the value without testing the `inherited` flag.